tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -211,43 +211,43 @@ def get_email_to_addrs_to_resources_map(self, sqs_message):
         # eg: { ('milton@initech.com', 'peter@initech.com'): [resource1, resource2, etc] }
-        return email_to_addrs_to_resources_map
+        return emails_to_resources_map
 
-    def get_to_addrs_email_messages_map(self, sqs_message):
-        to_addrs_to_resources_map = self.get_email_to_addrs_to_resources_map(sqs_message)
-        to_addrs_to_mimetext_map = {}
-        for to_addrs, resources in to_addrs_to_resources_map.items():
-            to_addrs_to_mimetext_map[to_addrs] = get_mimetext_message(
+    def get_emails_to_mimetext_map(self, sqs_message):
+        emails_to_resources_map = self.get_emails_to_resources_map(sqs_message)
+        emails_to_mimetext_map = {}
+        for to_addrs, resources in emails_to_resources_map.items():
+            emails_to_mimetext_map[to_addrs] = get_mimetext_message(
                 self.config, self.logger, sqs_message, resources, list(to_addrs)
             )
         # eg: { ('milton@initech.com', 'peter@initech.com'): mimetext_message }
-        return to_addrs_to_mimetext_map
+        return emails_to_mimetext_map
 
-    def send_c7n_email(self, sqs_message, email_to_addrs, mimetext_msg):
+    def send_c7n_email(self, sqs_message):
+        emails_to_mimetext_map = self.get_emails_to_mimetext_map(sqs_message)
+        email_to_addrs = list(emails_to_mimetext_map.keys())
         try:
             # if smtp_server is set in mailer.yml, send through smtp
             if "smtp_server" in self.config:
-                smtp_delivery = SmtpDelivery(
-                    config=self.config, session=self.session, logger=self.logger
-                )
-                smtp_delivery.send_message(message=mimetext_msg, to_addrs=email_to_addrs)
+                delivery = SmtpDelivery(self.config, self.session, self.logger)
+                for emails, mimetext_msg in emails_to_mimetext_map.items():
+                    email_to_addrs = list(emails)
+                    delivery.send_message(message=mimetext_msg, to_addrs=email_to_addrs)
             elif "sendgrid_api_key" in self.config:
-                sendgrid_delivery = sendgrid.SendGridDelivery(
-                    config=self.config, session=self.session, logger=self.logger
-                )
-                sendgrid_delivery.sendgrid_handler(
-                    sqs_message, self.get_to_addrs_email_messages_map(sqs_message)
-                )
+                delivery = SendGridDelivery(self.config, self.session, self.logger)
+                delivery.sendgrid_handler(sqs_message, emails_to_mimetext_map)
             # if smtp_server or sendgrid_api_key isn't set in mailer.yml, use aws ses normally.
             else:
-                self.aws_ses.send_raw_email(RawMessage={"Data": mimetext_msg.as_string()})
+                for emails, mimetext_msg in emails_to_mimetext_map.items():
+                    email_to_addrs = list(emails)
+                    self.aws_ses.send_raw_email(RawMessage={"Data": mimetext_msg.as_string()})
```

Comment:
Related to the comment above... we don't use `email_to_addrs` for the SSM send, so it feels clearer to leave out.

```suggestion
                for mimetext_msg in emails_to_mimetext_map.values():
                    self.aws_ses.send_raw_email(RawMessage={"Data": mimetext_msg.as_string()})
```